src/ctia/import/threatgrid/ttps.clj

```diff
@@ -0,0 +1,78 @@
+(ns ctia.import.threatgrid.ttps
+  (:require [clojure.data.json :as json]
+            [cheshire.core :as cjson]
+            [clojure.java.io :as io]
+            [clojure.pprint :refer [pprint]]
+            [ctim.schemas.ttp :as ttp]
+            [schema.core :as s]
+            [ctia.import.threatgrid.http :as http]
+            [ctia.lib.time :as time]))
+
+(s/defschema TGTTP
+  {(s/required-key "id") s/Str
+   (s/required-key "title") s/Str
+   (s/required-key "description") s/Str
+   ;;(s/required-key "short_description") s/Str
+   (s/required-key "timestamp") s/Str
+   (s/required-key "type") s/Str
+   (s/required-key "expires") s/Str
+   (s/required-key "indicators") [(s/maybe s/Str)]})
+
+(defn tg-ttp->ctim-ttp [validate-tg-ttp-fn validate-ctim-ttp-fn]
+  (fn [{:strs [id title description timestamp type expires indicators]
+        :as tg-ttp}]
+    (validate-tg-ttp-fn tg-ttp)
+    (validate-ctim-ttp-fn
+     (-> {:title id
```

Comment:
I would suggest to use a `s/defn` here to remove the manual use of the validate.
And if you want to activate/disable the validation you'll only have to call

~~~clojure
(s/set-fn-validation! true)
~~~ 

I believe you simply need to write something like:

~~~clojure
(defn tg-ttp->ctim-ttp :- [NewTTP]
   [{:strs [id title description timestamp type expires indicators]
     :as tg-ttp} :- TGTTP]
   ...
~~~